modules/sdp/SDP.js

```diff
@@ -136,6 +200,96 @@ export default class SDP {
         });
 
         this.raw = this.session + this.media.join('');
+
+        if (this.isP2P) {
+            return;
+        }
+
+        // For offers from Jicofo, a new m-line needs to be created for each new remote source that is added to the
+        // conference.
+        const newSession = transform.parse(this.raw);
+        const newMedia = [];
+
+        newSession.media.forEach(mLine => {
+            const type = mLine.type;
+
+            if (type === MediaType.APPLICATION) {
+                const newMline = cloneDeep(mLine);
+
+                newMline.mid = newMedia.length.toString();
```

Comment:
Is this what we really want? The lengfth as a string? Not its index?

Reply:
Yes, that is the requirement. The `mid` for the new m-line needs to be a string.

Reply:
I mean for the MID to the the length. I guess it's the index for the last added element, right?

Reply:
Yes.